external/os-guides/pentesting/kali.md

```diff
@@ -6,595 +6,156 @@ redirect_from:
+5. Copy the Kali PGP key from the DisposableVM to the new template:
 
-6. Copy the Kali GPG key from the DispVM to the new template:
+```shell_session
+$ qvm-copy kali-key.asc
+```
 
-        [user@xxxx-dvm ~]$ qvm-copy-to-vm kali-rolling kali-key.asc
+   The DisposableVM can now be turned off.
 
-   The DispVM can now be turned off.
+6. Add the Kali PGP key to the list of keys trusted to authenticate packages:
 
-7. Add the Kali GPG key to the list of keys trusted to authenticate packages:
+```shell_session
+# cat /home/user/QubesIncoming/dispXXX/kali-key.asc | apt-key add -
+```
 
-        [user@kali-rolling ~]$ cat /home/user/QubesIncoming/dispXXX/kali-key.asc | sudo apt-key add -
+   This command should return: `OK`.
 
-   This command should return `OK` on a line by itself.
+7. Replace Debian repositories with Kali repository
 
-8. Attempt the upgrade process in the new template.
+```shell_session
+# echo 'deb https://http.kali.org/kali kali-rolling main non-free contrib' > /etc/apt/sources.list
+```
 
-        [user@kali-rolling ~]$ sudo cat <<EOF > /etc/apt/sources.list.d/kali.list
-        # Kali Linux repository
-        deb http://http.kali.org/kali kali-rolling main non-free contrib
-        EOF
-        [user@kali-rolling ~]$ sudo apt-get update
-        [user@kali-rolling ~]$ sudo apt-get dist-upgrade
-        [user@kali-rolling ~]$ sudo apt-get autoremove
- 
-9. Shut down the new template.
+8. Replace `gcc8` to work around a dependency issue
 
-        [user@dom0 ~]$ qvm-shutdown kali-rolling
+```shell_session
+# apt-get remove libgcc-8-dev && apt-get install libc6-dev
+```
 
-10. Ensure a terminal can be opened in the new template.
-
-        [user@dom0 ~]$ qvm-run -a kali-rolling gnome-terminal
-
-#### Install the Kali tools
-
-At this point you should have a working template and you can install the tools you need.
-
-1. [Resize the template disk image][qubes-resize-disk-image] if you plan on installing the full Kali distribution. For example to install `kali-linux-full` you must **grow** the size of the VM system from 10GB to at least 20GB.
-
-2. Install Kali Linux tools:
-
-        [user@kali-rolling ~]$ sudo apt-get install kali-linux-full
-        
-3. (Optional) Customise the template's home directory (e.g. install your licensed copy of Burp Suite Professional)
-        
-#### Use the template
-
-The template is ready to be used. You can now spin up AppVMs based on the `kali-rolling` template.
-
-
-## Qubes 4.0
-
-### How to Create a Kali Linux VM
+**Note:** This kind of dependency issues will pop up and disappear without notice.
+It depends on the differences of dependencies between the Kali repository, the testing Qubes repository and the dependency issues related to the upgrade from stable Debian to testing Debian/Kali.
```

Comment:
What about:
Such issues arise because of the differences of dependencies in packages from the Kali repository, the Qubes testing repository and the Debian testing repository.

Reply:
Thanks, replaced the content